src/App.js

```diff
@@ -1,22 +1,27 @@
-/* eslint-disable import/extensions */
-import { useSelector } from 'react-redux';
-import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import React, { useEffect } from 'react';
 import { Route, Routes } from 'react-router-dom';
 import './index.css';
 import Books from './components/books';
 import Navbar from './components/navbar';
 import Categories from './components/categories';
+import { getBooks } from './redux/books/books';
 
 function App() {
-  const state = useSelector((state) => state);
+  const { books, categories } = useSelector((state) => state);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getBooks());
+  }, [books, dispatch]);
+
```

Comment:
Currently, the delete operation does not work. There's some issue: the `getBooks` action never stops dispatching. First, I think that this action is better dispatched within the `books` component. Also, the array argument should be empty since we only want to dispatch the action when the page loads.